examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1730,6 +1788,55 @@ class GLTFWriter {
 
 	}
 
+	/**
+	 * If a vertex attribute with a
+	 * [non-standard data type](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes-overview)
+	 * is used, then it is assumed that this is an allowed data type according to the
+	 * [KHR_mesh_quantization](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_mesh_quantization/README.md)
+	 * extension.
+	 * In this case the extension is automatically added to the list of used extensions.
+	 *
+	 * @param {string} attributeName
+	 * @param {THREE.BufferAttribute} attribute
+	 */
+	detectMeshQuantization( attributeName, attribute ) {
+
+		if ( this.extensionsUsed[ KHR_MESH_QUANTIZATION ] ) return;
+
+		let useMeshQuantization = false;
+
+		const isNonFloatAttribute = attribute.array.constructor !== Float32Array;
+
+		if ( isNonFloatAttribute ) {
+
+			if ( [ 'POSITION', 'NORMAL', 'TANGENT' ].includes( attributeName ) ) {
+
+				useMeshQuantization = true;
+
+			} else if ( attributeName.startsWith( 'TEXCOORD_' ) ) {
+
+				const isUnsignedByte = attribute.array.constructor === Uint8Array;
+				const isUnsignedShort = attribute.array.constructor === Uint16Array;
+
+				if ( ! ( attribute.normalized && ( isUnsignedByte || isUnsignedShort ) ) ) {
+
+					useMeshQuantization = true;
+
+				}
```

Comment:
I think it'd be clearer if we checked what the array is, rather than what it is not, here — perhaps:

```javascript
if ( attribute.normalized && ( isByte || isShort ) ) {

  useMeshQuantization = true;

}
```

Reply:
absolutely agree with you that a negation is difficult to understand here, but a positive formulation would have to cover much more cases and thus become more complex. so i have rethought the whole thing in detail and reworked the check as a whole.... hope it is more readable this way, in any case it is more correct and strictly adheres to the specification of the KHR_mesh_quantization extension